pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/NamespacesBase.java

```diff
@@ -1171,6 +1173,51 @@ protected void internalSplitNamespaceBundle(AsyncResponse asyncResponse, String
         });
     }
 
+    protected TopicHashPositions internalGetTopicHashPositions(String bundleRange, List<String> topicList) {
+        if (log.isDebugEnabled()) {
+            log.debug("[{}] Getting hash position for topic list {}, bundle {}", clientAppId(), topicList, bundleRange);
+        }
+        if (topicList == null || topicList.size() == 0) {
+            return null;
+        }
+        validateNamespacePolicyOperation(namespaceName, PolicyName.PERSISTENCE, PolicyOperation.READ);
+        Policies policies = getNamespacePolicies(namespaceName);
+        NamespaceBundle bundle = validateNamespaceBundleOwnership(namespaceName, policies.bundles, bundleRange,
+                false, true);
+        try {
+            List<String> allTopicsInThisBundle =
+                    pulsar().getNamespaceService().getOwnedTopicListForNamespaceBundle(bundle).get();
+            if (allTopicsInThisBundle.size() == 0) {
+                return null;
+            }
+            Map<String, Long> topicHashPositions = new HashMap<>();
+            for (String topic : topicList) {
+                // partitioned topic
+                if (TopicName.get(topic).getPartitionIndex() == -1) {
+                    allTopicsInThisBundle.stream()
+                            .filter(t -> TopicName.get(t).getPartitionedTopicName()
+                                    .equals(TopicName.get(topic).getPartitionedTopicName()))
+                            .forEach(partition -> {
+                                topicHashPositions.put(partition,  pulsar().getNamespaceService()
+                                        .getNamespaceBundleFactory().getLongHashCode(partition));
+                            });
+                } else { // topic partition
+                    if (allTopicsInThisBundle.contains(topic)) {
```

Comment:
Should convert to TopicName first, the topic name might  be a short topic name

Reply:
sure